### ISSUE TITLE
CompositeMetric not working as expected

### ISSUE THREAD

**Issue by waihongON** [jdb78/pytorch-forecasting#946]
- PyTorch-Forecasting version: 
- PyTorch version: 0.10.1
- Python version: 3.7.4
- Operating System: Ubuntu 20.04

### Expected behavior

I was using the `CompositeLoss` module by adding `SMAPE` and `MAPE` losses, I expected the losses produced by the `CompositeLoss` to match the loss produced when evaluating `SMAPE` and `MAPE` metrics separately and adding the loss tensors. (see code to reproduce the problem)

### Actual behavior

In practice, the losses do not match. The printout from the code snippet below shows this when run on some random inputs:

```
cl = 1.8917236328125, ml + sl = 1.8917236328125
cl = 2.1996891498565674, ml + sl = 2.5076541900634766
cl = 2.889475107192993, ml + sl = 4.269047737121582
```

They are far from matching each other

### Code to reproduce the problem

```
from pytorch_forecasting import MAPE, SMAPE
import torch

# MAPE loss, SMAPE loss, and composite loss of MAPE + SMAPE
ml_fn = MAPE()
sl_fn = SMAPE()
cl_fn = MAPE() + SMAPE()

# some dummy data
batches = 3
batch_size = 10
length = 10
y_preds = [torch.rand(batch_size, length, dtype=torch.float) for _ in range(batches)]
ys = [torch.rand(batch_size, length, dtype=torch.float) for _ in range(batches)]

# evaluating losses on the data
for y_pred, y in zip(y_preds, ys):
    ml = ml_fn(y_pred, y)
    sl = sl_fn(y_pred, y)
    cl = cl_fn(y_pred, y)
    print(f'cl = {cl}, ml + sl = {ml + sl}')

```

I encountered a similar issue before with `MultiLoss` and the fix was to re-implement the `reset` method to reset the constituent metrics - I can confirm this fixes things for `CompositeLoss` as well.

One thing I'd like to point out though - is that Torchmetrics already has a similar loss module `CompositionalLoss` is there any effort to unify the two for pytorch forecasting?
